### PR TITLE
test(refactor): add runtime blocker edge receipts (#8464)

### PR DESCRIPTION
**Pull request by EffortlessSteven**
Problem: Runtime refactor blocker receipts covered exact static plans, but dynamic-boundary and generated-member blockers were only proven in core cutover receipts.
Fix: Add runtime rename/safe-delete receipt tests for eval-string dynamic boundaries and Moo generated members, and refresh provider/UX status wording without changing live refactor behavior.
Verification: `cargo test -p perl-lsp-rs --lib --profile agent --locked refactor_runtime_blocker -- --nocapture --test-threads=1`; `cargo test -p perl-lsp-rs-core --profile agent --locked rename_shadow -- --nocapture --test-threads=2`; `cargo test -p perl-lsp-rs-core --profile agent --locked safe_delete_shadow -- --nocapture --test-threads=2`; `cargo test -p xtask --profile agent --locked semantic_shadow_compare -- --nocapture`; `cargo xtask semantic-shadow-compare --check`; `cargo xtask semantic-scorecard --check`; `cargo xtask metrics parser-accuracy --check`; `cargo xtask update-status --only parser --check`; `cargo xtask metrics ratchet-check parser_accuracy`; `cargo clippy --locked --lib -p perl-lsp-rs --profile agent -- -D warnings -A missing_docs`; `cargo clippy --locked --lib -p perl-lsp-rs-core --profile agent -- -D warnings -A missing_docs`; `cargo xtask fmt --check`; `git diff --check`

### CODE REVIEW

File: crates/perl-lsp-rs/src/runtime/language/refactor_runtime_blocker_tests.rs [hunk 1]
```diff
@@ -160,3 +202,139 @@ fn refactor_runtime_blocker_ux_safe_delete_receipt_records_exact_static_plan()
...
+    let compiler = compiler_receipt(&runtime_receipt)?;
+
+    assert_eq!(runtime_receipt.get("provider").and_then(Value::as_str), Some("rename"));
+    assert_eq!(runtime_receipt.get("no_live_behavior_change").and_then(Value::as_bool), Some(true));
+    assert!(trace_count(compiler)? > 0, "rename receipt must carry fact-source traces");
+    assert_note_contains(
+        compiler,
+        &[
+            "rename runtime blocker UX",
+            "blocker_count=2",
```
[the remark this answers is not in the document]
Comment (EffortlessSteven) [reply crates/perl-lsp-rs/src/runtime/language/refactor_runtime_blocker_tests.rs]: Addressed in f35446ca2 by making the dynamic-boundary runtime receipt assertions count-agnostic. The receipt contract here is that a DynamicBoundary blocker is present, confirmation is required, and no live refactor behavior changes; rename may surface multiple dynamic-boundary blockers while safe-delete has one.
